src/MigrateOriginalMessageCalls/MigrateOriginalMessageCallsCommand.php

```diff
@@ -0,0 +1,129 @@
+ * @see       https://github.com/zendframework/zend-expressive-tooling for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-tooling/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Expressive\Tooling\MigrateOriginalMessageCalls;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Zend\Expressive\Tooling\ErrorConsoleTrait;
+
+class MigrateOriginalMessageCallsCommand extends Command
+{
+    use ErrorConsoleTrait;
+
+    const DEFAULT_SRC = '/src';
+
+    const HELP = <<< 'EOT'
+Migrate an Expressive application to remove calls to legacy
+request/response methods.
+
+Scans all PHP files under the --src directory for any calls to
+getOriginalRequest(), getOriginalUri(), or getOriginalResponse().
+
+In the case of getOriginalResponse(), the call will be rewritten to
+getAttribute('originalRequest', {requestVariable}).
+
+In the case of getOriginalUri(), the call will be rewritten to
+getAttribute('originalUri', {requestVariable}->getUri()).
+
+If any getOriginalResponse() calls are detected, the script will present
+a warning indicating the file(s) and detail how to correct these
+manually.
+EOT;
+
+    const HELP_OPT_SRC = <<< 'EOT'
+Specify a path to PHP files to scan for
+getOriginalRequest/getOriginalUri/getOriginalResponse calls. If not
+specified, assumes src/ under the current working path.
+EOT;
+
+    const TEMPLATE_RESPONSE_DETAILS = <<< 'EOT'
+To correct these files, look for a request argument, and update the
+following calls ($response may be a different variable):
+
+    $response->getOriginalResponse()
+
+to:
+
+    $request->getAttribute('originalResponse', $response)
+
+($request may be a different variable)
+EOT;
+
+    /**
+     * @var string Path from which to resolve default src directory
+     */
+    public $projectDir = '.';
+
+    /**
+     * Configure the console command.
+     */
+    protected function configure()
+    {
+        $this->setDescription('Migrate getOriginal*() calls to request attributes.');
+        $this->setHelp(self::HELP);
+        $this->addOption('src', 's', InputOption::VALUE_REQUIRED, self::HELP_OPT_SRC);
+    }
+
+    /**
+     * Execute console command.
+     *
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int Exit status
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        try {
+            $src = $this->getSrcDir($input);
+        } catch (ArgvException $e) {
+            $console = $this->getErrorConsole($output);
+            $console->writeln(sprintf(
+                '<error>Unable to determine src directory: %s</error>',
+                $e->getMessage()
+            ));
+            return 1;
```

Comment:
Let exceptions bubble up. If the exception message is meaningful, no additional code is needed